libraries/provider_slave_grants.rb

```diff
@@ -0,0 +1,45 @@
+class Chef
+  class Provider
+    # Used to setup permissions for MySQL replication user on master server
+    class MysqlmSlaveGrants < Chef::Provider::LWRPBase
+      use_inline_resources if defined?(use_inline_resources)
+
+      def whyrun_supported?
+        true
+      end
+
+      def action_create
+        grant_slave_privs
```

Comment:
In this case should we just follow the database cookbook pattern (with mysql2 gem)? https://github.com/opscode-cookbooks/database/blob/master/libraries/provider_database_mysql_user.rb#L158-L165 ?